client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java

```diff
@@ -223,24 +223,24 @@ public ServerListManager(Properties properties) throws NacosException {
     }
     
     private String initServerName(Properties properties) {
-        String serverName = "";
+        String serverName;
         //1.user define server name.
         if (properties != null && properties.containsKey(PropertyKeyConst.SERVER_NAME)) {
             serverName = properties.get(PropertyKeyConst.SERVER_NAME).toString();
         } else {
             // if fix url,use fix url join string.
             if (isFixed) {
                 serverName = FIXED_NAME + "-" + (StringUtils.isNotBlank(namespace) ? (StringUtils.trim(namespace) + "-")
-                        : "") + getFixedNameSuffix(serverUrls.toArray(new String[serverUrls.size()]));
+                        : "") + getFixedNameSuffix(serverUrls.toArray(new String[0]));
```

Comment:
why 0 length?

Reply:
Don't need to specify the array length if the ArrayList.length equal the Array.length. see JDK api List.toArray()